packages/@ngtools/json-schema/src/schema-tree.spec.ts

```diff
@@ -55,12 +55,12 @@ describe('@ngtools/json-schema', () => {
       });
 
       expect(proto.a instanceof Array).toBe(true);
-      expect(proto.a).toEqual([undefined, 'v1', undefined, 'v3']);
+      expect(proto.a).toEqual(['v1', 'v3']);
 
       // Set it to a string, which is valid.
       proto.a[0] = 'v2';
-      proto.a[1] = 'INVALID';
-      expect(proto.a).toEqual(['v2', undefined, undefined, 'v3']);
+      proto.a[1] = 'v3';
+      expect(proto.a).toEqual(['v2', 'v3']);
```

Comment:
You're not testing anywhere that setting `INVALID` would throw an error. You should.